model/common/src/icon4py/model/common/model_backends.py

```diff
@@ -51,6 +51,7 @@ def make_custom_dace_backend(
     async_sdfg_call: bool = True,
     optimization_args: dict[str, typing.Any] | None = None,
     use_metrics: bool = True,
+    use_zero_origin: bool = False,
```

Comment:
This change leads to a performance regression in `compute_theta_rho_face_values_and_pressure_gradient_and_update_vn`.

- @edopao is looking into a cleanup or ranges -> origin, length
- we could enable use_zero_origin only for the program where it is needed